File: assignment/src/main/java/com/guigaspar/assignment/services/CustomerService.java
package com.guigaspar.assignment.services;

import java.util.List;

import com.guigaspar.assignment.dtos.CustomerDTO;

public interface CustomerService {
	
	public CustomerDTO saveUser(CustomerDTO customerDto);
	public CustomerDTO getCustomerByID(Long customerId);
	public List<CustomerDTO> getAllCustomers();
	
}

File: assignment/src/main/resources/static/js/app/CustomerService.js
'use strict'

angular.module('gui.services', []).factory('CustomerService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getCustomerById = function(customerId) {
				var url = "/customers";
				return $http.get(url);
			}
			service.getAllCustomers = function() {
				return $http.get("/customers");
			}
			service.saveCustomer = function(customerDto) {
				return $http.post("/customers", customerDto);
			}
			return service;
		} ]);